rest/src/main/java/com/redhat/ipaas/rest/v1/controller/handler/rtdb/RealTimeDB.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.ipaas.rest.v1.controller.handler.rtdb;
+
+import javax.ws.rs.core.StreamingOutput;
+import java.io.*;
+
+/**
+ * Provides a Java API to read/update a Key/Value database presented to the user a
+ * single large persistent JSON tree/document.  You can read or update subsets of the
+ * the document by specifying the path to the nodes in the JSON document that your
+ * interested in working with.
+ */
+public interface RealTimeDB {
```

Comment:
I think it will be better do the dao interface refactoring in a separate PR.  This one is already getting quite large.  I'm move the jsondb specific DAO into a sub package to keep them grouped and add Spring conditional configs to them so they can be activated or not via config.